test/common/http/http2/codec_impl_test.cc

```diff
@@ -118,12 +130,11 @@ class Http2CodecImplTest : public testing::TestWithParam<Http2SettingsTestParam>
       ENVOY_LOG_MISC(error, "data size too big or too small");
       return;
     }
-    uint8_t buf[METADATA_MAX_PAYLOAD_SIZE] = {0};
-    data.copyOut(0, length, static_cast<void*>(buf));
+    data.copyOut(0, length, static_cast<void*>(buf_tmp));
     data.drain(length);
     // Keeps the frame header (9 bytes) valid, and corrupts the payload.
-    buf[10] |= 0xff;
-    data.add(buf, length);
+    buf_tmp[10] |= 0xff;
```

Comment:
Not sure I understand this change?

Reply:
To make buf_tmp a static member, we only need to allocate the memory once. I think, previously, we may need to allocate the memory every time the function is called.